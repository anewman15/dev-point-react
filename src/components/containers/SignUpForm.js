/* eslint-disable jsx-a11y/no-autofocus */
import { useState } from 'react';
import createUser from '../../sandbox/createUser';
import FormErrorsSection from './FormErrorsSection';
import SignUpSuccess from './SignUpSuccess';

const SignUpForm = () => {
  const userInfoInit = {
    email: '', username: '', password: '', password_confirmation: '',
  };

  const [userInfo, setUserInfo] = useState(userInfoInit);
  const [formErrors, setFormErrors] = useState({});
  const [formSuccess, setFormSuccess] = useState(false);

  const handleChange = e => {
    setUserInfo({
      ...userInfo,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = e => {
    e.preventDefault();
    createUser(userInfo)
      .then(response => {
        let res;
        if (response.status === 200) {
          setUserInfo(userInfoInit);
          res = response.json();
        }
        if (response.status === 422) {
          res = response.json();
        }
        return res;
      })
      .then(data => {
        if (data.errors) {
          setFormErrors(data.errors);
          setFormSuccess(false);
        }
        if (data.message === 'User created successfully!') {
          setFormSuccess(true);
          setFormErrors({});
        }
      });
  };

  return (
    <div className="my-6 mx-2 columns is-centered">
      <div className="column is-half has-background-warning border-warning">
        <h1 className="is-size-3 has-text-weight-bold is-text-centered p-2 my-3">Create an Account</h1>
        {Object.keys(formErrors).length >= 1 && <FormErrorsSection formErrors={formErrors} />}
        {formSuccess && <SignUpSuccess />}
        <form className="p-4" onSubmit={handleSubmit}>
          <div className="field">
            <label className="label" htmlFor="email">
              Email
              <div className="control has-icons-left has-icons-right">
                <input
                  className="input is-success"
                  type="text"
                  name="email"
                  onChange={handleChange}
                  value={userInfo.email}
                  placeholder="email@emails.com"
                  required
                  autoFocus
                />
                <span className="icon is-small is-left">
                  <i className="fas fa-envelope" />
                </span>
              </div>
            </label>
          </div>
          <div className="field">
            <label className="label" htmlFor="username">
              Username
              <div className="control has-icons-left has-icons-right">
                <input
                  className="input is-success"
                  type="text"
                  name="username"
                  onChange={handleChange}
                  value={userInfo.username}
                  placeholder="Minimum 5 characters"
                  required
                />
                <span className="icon is-small is-left">
                  <i className="fas fa-user" />
                </span>
              </div>
            </label>
          </div>
          <div className="field">
            <label className="label" htmlFor="password">
              Password
              <div className="control has-icons-left has-icons-right">
                <input
                  className="input is-success"
                  type="password"
                  name="password"
                  onChange={handleChange}
                  value={userInfo.password}
                  placeholder="Minimum 6 characters"
                  required
                />
                <span className="icon is-small is-left">
                  <i className="fas fa-lock" />
                </span>
              </div>
            </label>
          </div>
          <div className="field">
            <label className="label" htmlFor="password_confirmation">
              Confirm password
              <div className="control has-icons-left has-icons-right">
                <input
                  className="input is-success"
                  type="password"
                  name="password_confirmation"
                  onChange={handleChange}
                  value={userInfo.password_confirmation}
                  placeholder="Your password again"
                  required
                />
                <span className="icon is-small is-left">
                  <i className="fas fa-lock" />
                </span>
              </div>
            </label>
          </div>
          <div className="control">
            <button type="submit" className="button is-success my-5">Sign up</button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default SignUpForm;
