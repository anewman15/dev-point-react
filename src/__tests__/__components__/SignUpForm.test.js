import { unmountComponentAtNode } from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SignUpForm from '../../components/containers/SignUpForm';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('SignUpForm', () => {
  it('renders the SignUpForm component to the dom', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <SignUpForm />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
