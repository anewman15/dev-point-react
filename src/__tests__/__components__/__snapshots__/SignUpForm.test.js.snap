// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm renders the SignUpForm component to the dom 1`] = `
<div
  className="my-12 mx-auto"
>
  <div
    className="mx-auto w-8/12 md:w-6/12 lg:w-5/12 bg-yellow-300 rounded"
  >
    <h1
      className="px-2 pt-8 pb-2 mb-3 text-3xl text-center text-green-800 font-bold"
    >
      Create an Account
    </h1>
    <div
      className="px-6 pb-2"
    >
      <hr
        className="border-1 border-primary-600 px-2"
      />
    </div>
    <div
      className="flex justify-center items-center"
    >
      <form
        className="px-4 mx-auto mb-8"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <label
            className="label"
            htmlFor="email"
          >
            Email
            <div
              className="control"
            >
              <input
                autoFocus={true}
                className="input"
                name="email"
                onChange={[Function]}
                placeholder="email@emails.com"
                required={true}
                type="text"
                value=""
              />
              <span
                className="icon"
              >
                <i
                  className="fas fa-envelope"
                />
              </span>
            </div>
          </label>
        </div>
        <div
          className="field"
        >
          <label
            className="label"
            htmlFor="username"
          >
            Username
            <div
              className="control"
            >
              <input
                className="input"
                name="username"
                onChange={[Function]}
                placeholder="Minimum 5 characters"
                required={true}
                type="text"
                value=""
              />
              <span
                className="icon"
              >
                <i
                  className="fas fa-user"
                />
              </span>
            </div>
          </label>
        </div>
        <div
          className="field"
        >
          <label
            className="label"
            htmlFor="password"
          >
            Password
            <div
              className="control"
            >
              <input
                className="input"
                name="password"
                onChange={[Function]}
                placeholder="Minimum 6 characters"
                required={true}
                type="password"
                value=""
              />
              <span
                className="icon"
              >
                <i
                  className="fas fa-lock"
                />
              </span>
            </div>
          </label>
        </div>
        <div
          className="field"
        >
          <label
            className="label"
            htmlFor="password_confirmation"
          >
            Confirm password
            <div
              className="control"
            >
              <input
                className="input"
                name="password_confirmation"
                onChange={[Function]}
                placeholder="Your password again"
                required={true}
                type="password"
                value=""
              />
              <span
                className="icon"
              >
                <i
                  className="fas fa-lock"
                />
              </span>
            </div>
          </label>
        </div>
        <div
          className="control"
        >
          <button
            className="btn btn-primary"
            type="submit"
          >
            Sign up
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
